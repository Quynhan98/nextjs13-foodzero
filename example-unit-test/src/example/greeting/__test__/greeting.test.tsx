import { render, screen } from '@testing-library/react';
import * as renderer from 'react-test-renderer';
import { Greeting } from '..';

describe('Test Greeting component', () => {
  test('Component Greeting renders Hello World! as a text', () => {
    // Arrange
    render(<Greeting />);

    //Act
    // ... nothing

    // Assert
    const helloWorldElement = screen.getByText('Hello World!');
    expect(helloWorldElement).toBeInTheDocument();
  });

  test('Component Greeting should render correctly', () => {
    const component = renderer.create(<Greeting />).toJSON();

    expect(component).toMatchSnapshot();
  });
});
