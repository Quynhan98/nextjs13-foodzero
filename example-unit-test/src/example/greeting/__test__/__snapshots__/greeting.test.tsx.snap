// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Greeting component Component Greeting should render correctly 1`] = `
<div>
  <h2>
    Hello World!
  </h2>
  <p>
    It's good to see you!
  </p>
</div>
`;
